.github/workflows/ci.yml

```diff
@@ -1,53 +1,44 @@
 name: CI
+
 on: [push, pull_request]
+
+env:
+  DEFAULT_NODE_VERSION: 22.x
+
 jobs:
 
   lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
-          path: '**/node_modules'
-          key: ${{ runner.os }}-lint-modules-${{ hashFiles('**/yarn.lock') }}
-      - uses: actions/setup-node@v2
-        with:
-          node-version: 14.x
-      - run: yarn install
+          node-version: ${{ env.DEFAULT_NODE_VERSION }}
+          cache: yarn
+      - run: yarn install --frozen-lockfile --ignore-scripts
       - run: yarn run lint
 
   test:
+    needs: lint
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest]
+        os:
+          - ubuntu-latest
         node-version:
-          - 14.x
-          - 16.x
           - 18.x
           - 20.x
           - 22.x
     steps:
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v2
+      - run: git config --global core.autocrlf input
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Ensure line endings are consistent
-        run: git config --global core.autocrlf input
-      - name: Check out repository
-        uses: actions/checkout@v2
-      - uses: actions/cache@v2
```

Comment:
Can we either keep the cache action, or enable cache in `actions/setup-node`?

Reply:
It is enabled, the diff just shows it after the 'deletions' (that have not all been deleted anyway) quite a bit down. Should I sort the `with` entries alphabetically?

Reply:
Oh, my bad. Nvm then!